MPSGameLift/Code/Source/Components/UI/UiGameLiftFlexMatchConnect.h

```diff
@@ -0,0 +1,118 @@
+
+        static constexpr char StatusMatchmakingTicketReceived[] = "Matchmaking Ticket: %s.";
+        static constexpr char StatusMatchmakingMatchFound[] = "Match found!";
+        static constexpr char StatusMatchmakingFailedToReceiveTicket[] = "Failed to receive matchmaking ticket.\nAre server fleets running?";
+        static constexpr char StatusMatchmakingFailedToReceiveStatusUpdate[] = "Failed to receive status update.\nMake sure you are connected to the internet and try again.";
+        static constexpr char StatusMatchmakingTimedOut[] = "Matchmaking timed out!\nReport ticket id to a developer.";
+        static constexpr char StatusMatchmakingFailedReasonUnknown[] = "Matchmaking failed.\nReason unknown.";
+
+
+    public:
+        AZ_COMPONENT(MPSGameLift::UiGameLiftFlexMatchConnect, "{EFB9D394-8134-400F-B751-42BA81CD08A7}");
+
+        /*
+        * Reflects component data into the reflection contexts, including the serialization, edit, and behavior contexts.
+        */
+        static void Reflect(AZ::ReflectContext* context);
+
+    protected:
+        void Activate() override;
+        void Deactivate() override;
+    
+    private:
+        // AWSClientAuth::AWSCognitoAuthorizationNotificationBus::Handler overrides...
+        void OnRequestAWSCredentialsSuccess(const AWSClientAuth::ClientAuthAWSCredentials& awsCredentials) override;
+        void OnRequestAWSCredentialsFail(const AZStd::string& error) override;
+        
+        // RequestLatenciesCompleteEventHandler...
+        void OnRequestLatenciesComplete(const RegionalLatencies& regionLatencies);
+        RequestLatenciesCompleteEvent::Handler m_requestLatenciesComplete{ [this](const RegionalLatencies& regionLatencies) { OnRequestLatenciesComplete(regionLatencies); }};
+
+        // Listen for matchmaking events...
+        MatchmakingSuccessEvent::Handler m_onMatchmakingSuccess{ [this]()
+        {
+            PushStatusUpdate(StatusMatchmakingMatchFound);
+        } };
+        
+        MatchmakingFailedEvent::Handler m_onMatchmakingFailed{ [this](MatchmakingFailReason reason)
+        {
+            switch(reason)
+            {
+            case MatchmakingFailReason::FailedToReceiveTicket:
+                PushStatusFail(StatusMatchmakingFailedToReceiveTicket);
+                break;
+            case MatchmakingFailReason::FailedToReceiveStatusUpdate:
+                PushStatusFail(StatusMatchmakingFailedToReceiveStatusUpdate);
+                break;
+            case MatchmakingFailReason::TimedOut:
+                PushStatusFail(StatusMatchmakingTimedOut);
+                break;
+            default:
+                PushStatusFail(StatusMatchmakingFailedReasonUnknown);
+            }
+        } };
+
+        MatchmakingTicketReceivedEvent::Handler m_onMatchmakingTicketReceived{ [this](AZStd::string ticketId)
+        {
+            PushStatusUpdate(AZStd::string::format(StatusMatchmakingTicketReceived, ticketId.c_str()));
+        } };
+        
```

Comment:
Suggestion: Consider moving the initialization and lambdas into the constructor so that all this code can live in the cpp file instead of the header. The one-line handlers aren't so bad, but the m_onMatchmakingFailed is starting to get a little big for the implementation to live in the header.

Reply:
Thank you. fixed!